### ISSUE TITLE
onnxruntime-node "Error: Non tensor type is temporarily not supported."

### ISSUE THREAD

**Issue by mleyburnBV** [microsoft/onnxruntime#8007]
**Describe the bug**
I am using the onnxruntime-node npm package to attempt to run my onnx model. When I invoke the "session.run()" function, it returns the error "Error: Non tensor type is temporarily not supported.". I am creating a tensor in the format in which the model expects but still doesn't work. The error message surfaces here.
https://github.com/microsoft/onnxruntime/blob/009f342caf425edd1fe00832047d78d536d36402/js/node/src/tensor_helper.cc 

**System information**
- OS Platform and Distribution: Mac OS 11.3
- ONNX Runtime-Node version: 1.8.0

**To Reproduce**
- Below shows the JS code running the model and how the model was created in Python

***JS Code***
```bash
 const ort = require('onnxruntime-node')

 const session = await ort.InferenceSession.create('./test.onnx')

 const dataA = Float32Array.from([1, 1, 1, 1, 1, 1, 1, 1])
 const tensorA = new ort.Tensor('float32', dataA, [1,8])

 const feeds = { 'float_input': tensorA }

 const results = await session.run(feeds)
```

***How the model was created***
```bash
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType
import onnxruntime as rt

initial_type = [('float_input', FloatTensorType([1, 8]))]
onx = convert_sklearn(model, initial_types=initial_type)
with open("test.onnx", "wb") as f:
    f.write(onx.SerializeToString())
```

**Expected behaviour**
To run the model with the given inputs and return predictions as outputs.

**Urgency**
I have tight timelines regarding a fix for this problem, as we are trialing onnxjs as a possible solution for our use case.

Any help is greatly appreciated.

**Comment by mwillbanks** [comment]
Experiencing exactly the same issue, i think the above given sample is precisely similar to the model that I have.

**Comment by mwillbanks** [comment]
Alright our issue was in the outputs, it appears that the output couldn't map the following back: `sequence<map<int64,float32>` likely the sequence bit was the issue although it would literally be an array of arrays in this case.

**Comment by jordantoaster** [comment]
@mwil

> Alright our issue was in the outputs, it appears that the output couldn't map the following back: `sequence<map<int64,float32>` likely the sequence bit was the issue although it would literally be an array of arrays in this case.

Would you be able to give an example of this in code? I am not quite sure how to translate your answer into a solution.

**Comment by mwillbanks** [comment]
@jordantoaster well, it's a little tricky, so for right now we are not explicitly calling for that output which avoids the error.

First and foremost, we attempted to read the same output from python and then the same outputs in node.js.  Python worked, node.js did not.  The docs notate that it only allows for standard Tensor outputs.  We mapped that back and found that this one was the only output that was somewhat non-standard (likely due to the sequence type) which is running through ZipMap.

We evaluated the model file through https://netron.app/ which had shown the different outputs and the process that the model went through.  I know this is not precisely what you were asking for but hopefully it provides enough context.

**Comment by jordantoaster** [comment]
> @jordantoaster well, it's a little tricky, so for right now we are not explicitly calling for that output which avoids the error.
> 
> First and foremost, we attempted to read the same output from python and then the same outputs in node.js. Python worked, node.js did not. The docs notate that it only allows for standard Tensor outputs. We mapped that back and found that this one was the only output that was somewhat non-standard (likely due to the sequence type) which is running through ZipMap.
> 
> We evaluated the model file through https://netron.app/ which had shown the different outputs and the process that the model went through. I know this is not precisely what you were asking for but hopefully it provides enough context.

Thanks, the model is am using is showing similar behaviours to what you described above. If we only call for the output label, it works, but throws the error when we also try to get the probability. The type in our case is `sequence<map<string,float32>>`.

Hopefully someone from the dev team can shed some light on how we can resolve this issue, as I imagine that almost all users want the output probabilities.

**Comment by mwillbanks** [comment]
> Thanks, the model is am using is showing similar behaviours to what you described above. If we only call for the output label, it works, but throws the error when we also try to get the probability. The type in our case is `sequence<map<string,float32>>`.
> 
> Hopefully someone from the dev team can shed some light on how we can resolve this issue, as I imagine that almost all users want the output probabilities, it also exposes opportunities to improve error messages in the node run time.

💯 , my assumption at this point is that it has to do with [dtypes](https://www.tensorflow.org/api_docs/python/tf/dtypes/DType) and not handling more complex types than that.

**Comment by jordantoaster** [comment]
> > Thanks, the model is am using is showing similar behaviours to what you described above. If we only call for the output label, it works, but throws the error when we also try to get the probability. The type in our case is `sequence<map<string,float32>>`.
> > Hopefully someone from the dev team can shed some light on how we can resolve this issue, as I imagine that almost all users want the output probabilities, it also exposes opportunities to improve error messages in the node run time.
> 
> 💯 , my assumption at this point is that it has to do with [dtypes](https://www.tensorflow.org/api_docs/python/tf/dtypes/DType) and not handling more complex types than that.

We managed to find a resolution for this. Essentially it seems that ZipMap is the root cause of the problem. We changed out conversion code to the below:

```
initial_type = [('float_input', FloatTensorType((1, 8)))]
options = {id(model): {'zipmap': False}} # enables getting probabilities in Node.
onx = convert_sklearn(model, initial_types=initial_type, options=options)
```

I will leave this issue open for now, as I feel it is important that this needs to be documented **somewhere** for anybody attempting to convert from Python (SkLearn) and use in the ONNX NodeJS runtime.

**Comment by wanghsinche** [comment]
Please refer to this:  https://github.com/onnx/sklearn-onnx/issues/754#issuecomment-931397744        
 
It is caused by sklearn classifier output a object with data and probability. So the type casting can not work. set up zipmap to be false can make it work.

**Comment by guschmue** [comment]
I assume this is no longer an issue with 1.14.
